pkg/testsim/process.go

```diff
@@ -0,0 +1,123 @@
+// Copyright Contributors to the Mir project
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package testsim
+
+import (
+	"sync"
+	"time"
+)
+
+// Process represents a context of execution in the simulation.
+type Process struct {
+	*Runtime
+	lock     sync.Mutex
```

Comment:
Here, I actually prefer to use `Runtime` explicitly because it is not a part of `Process`, although it is conveniently declared as an embedded field, even exported for convenience of external packages. On the other hand, I don't want give external packages access to the lock, therefore made it an unexported field.